src/Sema.zig

```diff
@@ -17578,10 +17578,13 @@ pub fn analyzeAddrspace(
     const address_space = addrspace_tv.val.toEnum(std.builtin.AddressSpace);
     const target = sema.mod.getTarget();
     const arch = target.cpu.arch;
+    const is_gpu = arch == .nvptx or arch == .nvptx64;
 
     const supported = switch (address_space) {
         .generic => true,
         .gs, .fs, .ss => (arch == .i386 or arch == .x86_64) and ctx == .pointer,
+        .global, .shared, .local => is_gpu,
+        .constant => is_gpu and (ctx == .constant),
```

Comment:
I think that `constant` should also be valid for pointers. The context `.constant` refers to a global value that is constant in this case. Im also not sure if `global` should be valid for anything thats not a pointer. It should in theory be possible to declare a `local` and `shared` global value, but im not sure whether it can be initialized at all.

Reply:
followed up in the main thread